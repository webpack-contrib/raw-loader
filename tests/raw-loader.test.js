/* eslint-disable */
import fs from 'fs';
import { runLoaders } from 'loader-runner';
import Promise from 'bluebird';
import { getFixtureResource, getFixture, getLoader } from './test-utils';

const runLoadersPromise = Promise.promisify(runLoaders);
const readFilePromise = Promise.promisify(fs.readFile, { context: fs });

const loaders = getLoader();

describe('Raw Loader: Fixture: basic', () => {
  const fixtureName = 'basic.scss';

  beforeEach(() => {

  });

  test('should load scss file', async () => {
    const { result } = await runLoadersPromise({ resource: getFixtureResource(fixtureName), loaders });
    const expected = [ 'export default = ".another-scss-module {\\n  background: hotpink;\\n}"' ];

    expect(result).toEqual(expect.arrayContaining(expected));
    expect(result).toMatchSnapshot();
  });
});
