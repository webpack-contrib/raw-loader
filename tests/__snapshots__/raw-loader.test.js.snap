// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Raw Loader: Fixture: basic should load scss file 1`] = `
Array [
  "export default = \\".another-scss-module {\\\\n  background: hotpink;\\\\n}\\"",
]
`;
